Makefile

```diff
@@ -0,0 +1,84 @@
+# Makefile for the charm store client.
+
+ifndef GOPATH
+$(warning You need to set up a GOPATH.)
+endif
+
+PROJECT := github.com/juju/charmstore-client
+PROJECT_DIR := $(shell go list -e -f '{{.Dir}}' $(PROJECT))
```

Comment:
It is opaque but it basically is used to evaluate if go can find the project.

If it helps any, this is very well-worn trickery, cargo culted across most of our go projects.